### DISCUSSION TITLE
How to use Traefik reverse proxy with Nextcloud AIO?

### DISCUSSION THREAD

**Discussion by iamnikv** [nextcloud/all-in-one#622] [Questions] [answered]
I am following the guide for setting up Traefik and the command for running the docker container:

# For x64 CPUs:
sudo docker run -it \
--name nextcloud-aio-mastercontainer \
--restart always \
-p 8080:8080 \
-e APACHE_PORT=11000 \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /var/run/docker.sock:/var/run/docker.sock:ro \
nextcloud/all-in-one:latest

However when I get to here:

![image](https://user-images.githubusercontent.com/12248943/168135404-d520b998-4e03-453a-95cc-9fbf654fa683.png)

.. and enter my domain name I get this error:

![image](https://user-images.githubusercontent.com/12248943/168135499-8ebc6cdf-e9e3-4c7f-8561-3fbcf581af7f.png)

From the Traefik container:

![image](https://user-images.githubusercontent.com/12248943/168135887-3848d0a2-afd6-487b-8a53-c7ad051d62d3.png)

On the Nextcloud VM:

![image](https://user-images.githubusercontent.com/12248943/168136172-a7ae6c7a-14ea-47a2-ba23-f4676d39dd3d.png)

Traefik config:

http:
  routers:
![image](https://user-images.githubusercontent.com/12248943/168136447-5095e753-ea0e-4502-8544-25579c6bbeb2.png)
  services:
![image](https://user-images.githubusercontent.com/12248943/168136559-8edc99f4-b0b6-43d0-8968-ba6d6b043fbe.png)
![image](https://user-images.githubusercontent.com/12248943/168136627-2fbe1b43-61ab-4a84-92d9-996fc2109349.png)
![image](https://user-images.githubusercontent.com/12248943/168137049-9a59c385-8966-4bc8-a5e8-1d60b94bd4b3.png)

**Comment by szaimen** [comment]
Hi, can you run `nc -z <yourvalue> 11000` again from the traefik container with this value?
![image](https://user-images.githubusercontent.com/42591237/168140997-02654923-22ae-416a-a7d5-ba808e63f457.png)

**Reply by iamnikv** [reply]
Here is the result:

![image](https://user-images.githubusercontent.com/12248943/168158591-adfa739e-21f2-4ff9-bea9-f22229a62f06.png)

**Reply by szaimen** [reply]
And `echo $?`
Of that?

**Reply by iamnikv** [reply]
![image](https://user-images.githubusercontent.com/12248943/168159090-fd068f9d-b0d6-4a4b-bb1b-23377eb67463.png)

**Reply by szaimen** [reply]
Please run the nc -z command first and afterwards the echo $? command

**Reply by iamnikv** [reply]
Like this?

![image](https://user-images.githubusercontent.com/12248943/168159511-375705e4-6351-45c9-b5f4-7687d08d3a90.png)

**Reply by szaimen** [reply]
Yes. Looks good.

**Comment by szaimen** [comment]
I see what the problem is. Did you open port 443 in your firewall and router?

**Comment by iamnikv** [comment]
I have opened port 443 but for the VM that hosts the Traefik container. Should I open it for the Nextcloud VM as well?

**Reply by szaimen** [reply]
> I have opened port 443 but for the VM that hosts the Traefik container. Should I open it for the Nextcloud VM as well?

I don't understand. How can you open the same port mutiple times?

**Comment by szaimen** [comment, marked as answer]
Also is your domain correctly set up that you want to use for Nextcloud and does it point to your server?

**Reply by iamnikv** [reply]
I think it is correctly setup because I am using it for my other services and I can access them from outside of my local network by the domain name.

**Reply by szaimen** [reply]
The subdomain as well?

**Reply by iamnikv** [reply]
Ok I am dumb. I was pointing the domain name in the router config to the local ip of the Nextcloud VM instead of pointing it to the VM hosting the Traefik container. I changed it and not it works :) 
Sorry for the all of this and thank you for pointing me to the problem :)

**Reply by szaimen** [reply]
Great that you found the problem!

**Comment by szaimen** [comment]
Btw, did you see this?
https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md#traefik-2

**Reply by szaimen** [reply]
Just asking because your traefik config looks pretty different

**Reply by iamnikv** [reply]
Yes, I saw it and I compared the headers. Most of the I had already included. I had missing these two:
        hostsProxyHeaders = ["X-Forwarded-Host"]
        X-Robots-Tag = "none"
But wanted to see if it would work without them and it does.

**Comment by iamnikv** [comment]
For anyone who is trying to setup Traefik as reverse proxy I would like to link them to the guide I used for setting it up and it works really well for me. It the one from TechnoTim: https://techno-tim.github.io/posts/traefik-portainer-ssl/
